observ/proxy.py

```diff
@@ -21,12 +21,14 @@ class Proxy(Generic[T]):
     """
 
     __hash__ = None
-    __slots__ = ("target", "readonly", "shallow", "__weakref__")
+    # the slots have to be very unique since we also proxy objects
+    # which may define the attributes with the same names
+    __slots__ = ("__target__", "__readonly__", "__shallow__", "__weakref__")
```

Comment:
Since they need to be very unique, did you consider using a prefix such as `__observ_`?

Reply:
Yeah I did and it was really annoying to read...

Reply:
I can imagine that :) Let's keep it like it is now.